server/api/price_history_handler.py

```diff
@@ -0,0 +1,44 @@
+import json, re
+from flask import jsonify, request, Blueprint
+from models.prices import Prices
+from database import db
+price_history_handler = Blueprint('price_history_handler', __name__)
+from .scraper import ScrapeAmazon
+
+
+def get_url_id(URL):
+    x = re.search(r"amazon.com\/.*dp\/([^/]+)", URL) 
+    return x.group(1)
+    
+
+@price_history_handler.route('/price_history', methods = ['GET', 'POST'])
+def price_history():
+    if request.method == 'GET':
+        price_history = Prices.query.all()
+        price_history_ls = []
+        for item in price_history:
+            price_history_ls.append(
+                {
+                    "id": item.id,
+                    "url_id": item.url_id,
```

Comment:
what is `url_id` is that `product_id` , you don't technically need the `url_id` unless it's that case in which the product will be changing urls, in which case the user would need to input the new url

Reply:
sorry, I just realized, you tried to do a regex, to get the url id based on the amazon link

Reply:
this still doesn't make sense because we still need a way to link between the product table and the `prices_history` table, that's why we need `product_id`. I think the scrape URL code makes sense maybe in the scraper code to check if it's an amazon link?